### PR TITLE
Carthage qn categories metaprotocol 2

### PR DESCRIPTION
**Pull request by ondratra**
Fixes https://github.com/Joystream/joystream/issues/3574.
Superseeds https://github.com/Joystream/joystream/pull/3950. Builds on top of https://github.com/Joystream/joystream/pull/4021 and expects it to be merged first.

This PR contains:
- removal of channel categories from QN
- rework of video categories - from Joystream-node-based event's to metaprotocol-based representation
- adjustments to comments and reactions code so it can use common parts with video categories
- lint checks improvements
- reenabling of active video counter integration test

### CODE REVIEW

**Comment by ondratra** [comment]
I'm not sure why is the the active-video-counters test is failing in CI.
When I run it locally via 
```
DEBUG=true ./query-node/run-tests.sh # QN tests only
DEBUG=true ./query-node/run-tests.sh full # full tests
```
everything passes just fine.

**Comment by kdembler** [comment]
@ondratra @mnaamani I think we should add categories creation in chain setup script so new playgrounds have some categories from the start

File: tests/network-tests/src/scenarios/content-directory.ts [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
-  job('nft auction and offers', nftAuctionAndOffers).requires(channelJob)
-  job('curator moderation actions', curatorModerationActions).requires(channelJob)
-  job('video comments and reactions', commentsAndReactions).after(channelJob)
+  const videoCategoriesJob = job('video categories', videoCategories).after(channelJob)
+  const videoCountersJob = job('check active video counters', activeVideoCounters).requires(videoCategoriesJob)
+  const nftAuctionAndOffersJob = job('nft auction and offers', nftAuctionAndOffers).after(videoCountersJob)
+  const curatorModerationActionsJob = job('curator moderation actions', curatorModerationActions).after(
+    nftAuctionAndOffersJob
+  )
+  job('video comments and reactions', commentsAndReactions).after(curatorModerationActionsJob)
```
Comment (zeeshanakram3) [inline tests/network-tests/src/scenarios/content-directory.ts:22]: `curatorModerationActionsJob` does not really need `nftAuctionAndOffersJob` to be completed first. So they can be run in parallel. I think current approach unnecessarily increases the running time of tests
```suggestion
  job('nft auction and offers', nftAuctionAndOffers).after(videoCategoriesJob)
  job('curator moderation actions', curatorModerationActions).after(videoCategoriesJob)
  job('video comments and reactions', commentsAndReactions).after(videoCategoriesJob)
```

Comment (zeeshanakram3) [inline tests/network-tests/src/scenarios/content-directory.ts:22]: `curatorModerationActionsJob` does not really need `nftAuctionAndOffersJob` to be completed first. So they can be run in parallel. I think the current approach unnecessarily increases the running time of tests
```suggestion
  job('nft auction and offers', nftAuctionAndOffers).after(videoCategoriesJob)
  job('curator moderation actions', curatorModerationActions).after(videoCategoriesJob)
  job('video comments and reactions', commentsAndReactions).after(videoCategoriesJob)
```

File: tests/network-tests/src/cli/joystream.ts [hunk 2]
```diff
@@ -2,14 +2,22 @@ import { KeyringPair } from '@polkadot/keyring/types'
 import path from 'path'
 import { CLI, CommandResult } from './base'
 import { TmpFileManager } from './utils'
+import { Utils } from '../utils'
```
Comment (zeeshanakram3) [inline tests/network-tests/src/cli/joystream.ts:5]: Unused import
```suggestion
```

File: tests/network-tests/src/cli/joystream.ts [hunk 3]
```diff
@@ -2,14 +2,22 @@ import { KeyringPair } from '@polkadot/keyring/types'
...
 } from '@joystream/cli/src/Types'
 import ExitCodes from '@joystream/cli/src/ExitCodes'
+import {
+  CreateVideoCategory,
+  UpdateVideoCategory,
+  DeleteVideoCategory,
+  ModerateVideoCategories,
+  WorkerGroupLeadRemarked,
+} from '@joystream/metadata-protobuf'
+import Long from 'long'
```
Comment (zeeshanakram3) [inline tests/network-tests/src/cli/joystream.ts:20]: Unused imports
```suggestion
```

File: tests/network-tests/src/cli/joystream.ts [hunk 4]
```diff
@@ -2,14 +2,22 @@ import { KeyringPair } from '@polkadot/keyring/types'
 import path from 'path'
 import { CLI, CommandResult } from './base'
 import { TmpFileManager } from './utils'
+import { Utils } from '../utils'
 import {
   VideoInputParameters,
   ChannelCreationInputParameters,
   ChannelUpdateInputParameters,
-  ChannelCategoryInputParameters,
   VideoCategoryInputParameters,
```
Comment (zeeshanakram3) [inline tests/network-tests/src/cli/joystream.ts:10]: Unused import
```suggestion
```

File: tests/network-tests/src/fixtures/content/nft/englishAuction.ts [hunk 5]
```diff
@@ -8,9 +8,6 @@ import { Utils } from '../../../utils'
 import { assertNftOwner, assertAuctionAndBids, assertNftEventContentActor } from './utils'
 import BN from 'bn.js'
```
Comment (zeeshanakram3) [inline tests/network-tests/src/fixtures/content/nft/englishAuction.ts:9]: Unused import
```suggestion
```

File: tests/network-tests/src/fixtures/content/nft/openAuction.ts [hunk 6]
```diff
@@ -6,9 +6,6 @@ import { PlaceBidsInAuctionFixture } from './placeBidsInAuction'
 import { assertNftOwner, assertAuctionAndBids } from './utils'
 import BN from 'bn.js'
```
Comment (zeeshanakram3) [inline tests/network-tests/src/fixtures/content/nft/openAuction.ts:7]: Unused import
```suggestion
```

File: tests/network-tests/src/fixtures/content/createContentStructure.ts [hunk 7]
```diff
@@ -5,30 +5,25 @@ import { Api } from '../../Api'
 import { PalletWorkingGroupGroupWorker as Worker } from '@polkadot/types/lookup'
 import { getVideoCategoryDefaults, getChannelCategoryDefaults } from './contentTemplates'
```
Comment (zeeshanakram3) [inline tests/network-tests/src/fixtures/content/createContentStructure.ts:6]: unused import
```suggestion
import { getVideoCategoryDefaults } from './contentTemplates'
```

File: query-node/mappings/src/content/channel.ts [hunk 8]
```diff
@@ -1,17 +1,18 @@
 /*
 eslint-disable @typescript-eslint/naming-convention
 */
-import { DatabaseManager, EventContext, StoreContext } from '@joystream/hydra-common'
+import { DatabaseManager, EventContext, StoreContext, SubstrateEvent } from '@joystream/hydra-common'
 import { ChannelMetadata, ChannelModeratorRemarked, ChannelOwnerRemarked } from '@joystream/metadata-protobuf'
 import { BaseModel } from '@joystream/warthog'
```
Comment (zeeshanakram3) [inline query-node/mappings/src/content/channel.ts:6]: unused import
```suggestion
```

File: query-node/mappings/src/content/channel.ts [hunk 9]
```diff
@@ -1,17 +1,18 @@
...
+import { ChannelId, DataObjectId } from '@joystream/types/primitives'
 import {
   Channel,
+  ContentActor,
   ContentActorCurator,
   ContentActorMember,
   CuratorGroup,
   Membership,
   MetaprotocolTransactionErrored,
```
Comment (zeeshanakram3) [inline query-node/mappings/src/content/channel.ts:15]: unused import
```suggestion
```

File: query-node/mappings/src/content/channel.ts [hunk 10]
```diff
@@ -1,17 +1,18 @@
...
-  MetaprotocolTransactionPending,
   MetaprotocolTransactionStatusEvent,
```
Comment (zeeshanakram3) [inline query-node/mappings/src/content/channel.ts:16]: unused import
```suggestion
```

File: query-node/mappings/src/content/channel.ts [hunk 11]
```diff
@@ -23,11 +24,11 @@ import { In } from 'typeorm'
...
-  genericEventFields,
   inconsistentState,
-  invalidMetadata,
+  genericEventFields,
   logger,
-  updateMetaprotocolTransactionStatus,
+  saveMetaprotocolTransactionSuccessful,
+  saveMetaprotocolTransactionErrored,
 } from '../common'
 import { getAllManagers } from '../derivedPropertiesManager/applications'
```
Comment (zeeshanakram3) [inline query-node/mappings/src/content/channel.ts:33]: unused import
```suggestion
```

File: metadata-protobuf/proto/Metaprotocol.proto [hunk 12]
```diff
@@ -136,4 +136,31 @@ message ChannelOwnerRemarked {
...
+        UpdateVideoCategory update_category = 2;
+        DeleteVideoCategory delete_category = 3;
+    }
+}
+
+message WorkerGroupLeadRemarked {
+    oneof worker_group_lead_remarked {
+        ModerateVideoCategories moderate_video_categories = 1;
+    }
+}
```
Comment (zeeshanakram3) [inline metadata-protobuf/proto/Metaprotocol.proto:166]: The nested `oneof` message doesn't seem necessary here, we can achieve the purpose by only a single `oneof` message as well, i.e. directly wrapping `*VideoCategory` messages inside `WorkerGroupLeadRemarked`

The message serialized with the schema described in the PR would have a few extra bytes (needed to encode nested `oneof` field) compared to the schema proposed in this comment

Given that the serialized message would be part of on-chain transaction payload, and hence would impact transaction fees, we should  aim for minimum serialized message size
```suggestion
message WorkerGroupLeadRemarked {
    oneof worker_group_lead_remarked {
        CreateVideoCategory create_category = 1;
        UpdateVideoCategory update_category = 2;
        DeleteVideoCategory delete_category = 3;
    }
}
```

**Review by zeeshanakram3 — CHANGES_REQUESTED** [review]

**Comment by dmtrjsg** [comment]
Closing this PR as superceded by https://github.com/Joystream/joystream/pull/4122

**Pull request closed without merging**